pkg/artifactory/resource/configuration/resource_artifactory_mail_server.go

```diff
@@ -0,0 +1,338 @@
+	r.SubjectPrefix = types.StringValue(mailServer.SubjectPrefix)
+	r.UseSSL = types.BoolValue(mailServer.UseSSL)
+	r.UseTLS = types.BoolValue(mailServer.UseTLS)
+
+	return nil
+}
+
+func NewMailServerResource() resource.Resource {
+	return &MailServerResource{}
+}
+
+type MailServerResource struct {
+	ProviderData utilsdk.ProvderMetadata
+}
+
+func (r *MailServerResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "artifactory_mail_server"
+}
+
+func (r *MailServerResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Provides an Artifactory mail server config resource. This resource configuration corresponds to mail server config block in system configuration XML (REST endpoint: artifactory/api/system/configuration). Manages mail server settings of the Artifactory instance.",
+		Attributes: map[string]schema.Attribute{
+			"enabled": schema.BoolAttribute{
+				MarkdownDescription: "When set, mail notifications are enabled.",
+				Required:            true,
+			},
+			"artifactory_url": schema.StringAttribute{
+				MarkdownDescription: "The Artifactory URL to to link to in all outgoing messages.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					validatorfw_string.IsURLHttpOrHttps(),
+				},
+			},
+			"from": schema.StringAttribute{
+				MarkdownDescription: "The 'from' address header to use in all outgoing messages.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					validatorfw_string.IsEmail(),
+				},
+			},
+			"host": schema.StringAttribute{
+				MarkdownDescription: "The mail server IP address / DNS.",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"username": schema.StringAttribute{
+				MarkdownDescription: "The username for authentication with the mail server.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"password": schema.StringAttribute{
+				MarkdownDescription: "The password for authentication with the mail server.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"port": schema.Int64Attribute{
+				MarkdownDescription: "The port number of the mail server.",
+				Required:            true,
+			},
+			"subject_prefix": schema.StringAttribute{
+				MarkdownDescription: "A prefix to use for the subject of all outgoing mails.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"use_ssl": schema.BoolAttribute{
```

Comment:
I am pretty sure these 2 have to be xor. [TLS supercedes SSL](https://aws.amazon.com/compare/the-difference-between-ssl-and-tls/#:~:text=SSL%20is%20technology%20your%20applications,that%20fixes%20existing%20SSL%20vulnerabilities.)
```
 SSL is technology your applications or browsers may have used to create a secure, encrypted communication channel over any network. However, SSL is an older technology that contains some security flaws. Transport Layer Security (TLS) is the upgraded version of SSL that fixes existing SSL vulnerabilities. TLS authenticates more efficiently and continues to support
```

Reply:
Nope, Artifactory allows both to be set simultaneously.